src/python/tests/core/api/test_scheduler.py

```diff
@@ -0,0 +1,458 @@
+    @pytest.mark.asyncio
+    async def test_get_tasks(self):
+        # First create a schedule to get the schedule_id
+        headers = {"Content-Type": 'application/json'}
+        data = {"type": 3, "name": "test_get_task1", "process_name": "sleep1", "repeat": 2}
+
+        r = requests.post('http://localhost:8082/foglamp/schedule', data=json.dumps(data), headers=headers)
+        retval = dict(r.json())
+
+        assert 200 == r.status_code
+        schedule_id = retval['schedule']['id']
+
+        # Now start the schedule to create a Task
+        r = requests.post('http://localhost:8082/foglamp/schedule/start/' + schedule_id)
+        retval = dict(r.json())
+
+        assert 200 == r.status_code
+        assert retval['id'] == schedule_id
+        assert retval['message'] == "Schedule started successfully"
+
+        # Allow multiple task records to be created
+        await asyncio.sleep(10)
+
+        # Verify with Task record as to two  tasks have been created
+        rr = requests.get('http://localhost:8082/foglamp/task')
+        retvall = dict(rr.json())
+        print(retvall)
+
+        assert 200 == rr.status_code
+        assert len(retvall['tasks']) > 0
+        assert retvall['tasks'][0]['process_name'] == 'sleep1'
+        assert retvall['tasks'][1]['process_name'] == 'sleep1'
+
+
+    @pytest.mark.run(order=11)
+    @pytest.mark.asyncio
+    async def test_get_tasks_latest(self):
+        # First create a schedule to get the schedule_id
+        headers = {"Content-Type": 'application/json'}
+        data = {"type": 3, "name": "test_get_task1", "process_name": "sleep1", "repeat": 2}
+
+        r = requests.post('http://localhost:8082/foglamp/schedule', data=json.dumps(data), headers=headers)
+        retval = dict(r.json())
+
+        assert 200 == r.status_code
+        schedule_id = retval['schedule']['id']
+
+        await asyncio.sleep(4)
+
+        # Now start the schedule to create a Task
+        r = requests.post('http://localhost:8082/foglamp/schedule/start/' + schedule_id)
+        retval = dict(r.json())
+
+        assert 200 == r.status_code
+        assert retval['id'] == schedule_id
+        assert retval['message'] == "Schedule started successfully"
+
+        # Allow multiple tasks to be created
+        await asyncio.sleep(10)
+
+        # Verify with Task record as to more than one task have been created
+        r = requests.get('http://localhost:8082/foglamp/task')
+        retval = dict(r.json())
+        print(retval)
+
+        assert 200 == r.status_code
+        assert len(retval['tasks']) > 0
+
+        # Verify only one Task record is returned
+        r = requests.get('http://localhost:8082/foglamp/task/latest')
+        retval = dict(r.json())
+
+        assert 200 == r.status_code
+        assert 1 == len(retval['tasks'])
+        assert retval['tasks'][0]['process_name'] == 'sleep1'
+
+
+    @pytest.mark.run(order=12)
+    @pytest.mark.asyncio
+    async def test_cancel_task(self):
```

Comment:
we are testing cacel task, task creation etc. are already tested.. so make it specific.